File: codility/Dominator.js
function solution(A) {
    // Implement your solution here
    const obj = {};
    const half = A.length / 2;
    let result = -1;
    for (let i = 0; i < A.length; i++) {
        const v = A[i];
        if (obj[v]) {
            obj[v]++;
        } else {
            obj[v] = 1;
        }
        if (obj[v] > half) {
            result = i;
            break;
        }
    }
    // console.log( A, A.indexOf(Number(result)))
    return result;
}

File: codility/python/Dominator.py
def solution(A):
    # Implement your solution here
    dict = {}
    n = len(A)
    for i in range(n):
        a = A[i]
        dict[a] = dict[a]+1 if dict.get(a, None) != None else 1
        if dict[a] > n / 2:
            return i 
    return -1